From 97ac5c34369525cc53f264238fd2fb870f4f5933 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 28 Apr 2023 13:29:45 -0400
Subject: [PATCH 5/5] libxl: Switch to ndb+unix:/// format

Switch to the URI format specified by the protocol at
https://github.com/NetworkBlockDevice/nbd/blob/master/doc/uri.md

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/libs/light/libxl_disk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/libs/light/libxl_disk.c b/tools/libs/light/libxl_disk.c
index 525ae0db97..b65cad33cc 100644
--- a/tools/libs/light/libxl_disk.c
+++ b/tools/libs/light/libxl_disk.c
@@ -1343,7 +1343,7 @@ char *libxl__device_disk_find_local_path(libxl__gc *gc,
 
             /* tapdisk exposes disks locally over UNIX socket NBD. */
             if (disk->backend == LIBXL_DISK_BACKEND_TAP) {
-                path = libxl__sprintf(gc, "nbd:unix:%s", path);
+                path = libxl__sprintf(gc, "nbd+unix:///?socket=%s", path);
                 LOGD(DEBUG, guest_domid,
                      "Directly accessing local TAP target %s", path);
             }
-- 
2.43.0

